addOns/llm/src/main/java/org/zaproxy/addon/llm/services/LlmCommunicationService.java

```diff
@@ -0,0 +1,169 @@
+package org.zaproxy.addon.llm.services;
+
+import com.azure.ai.openai.models.ChatCompletionsJsonResponseFormat;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.azure.AzureOpenAiChatModel;
+import dev.langchain4j.service.AiServices;
+import org.apache.commons.httpclient.util.HttpURLConnection;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.db.DatabaseException;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.zaproxy.addon.llm.communication.Confidence;
+import org.zaproxy.addon.llm.communication.HttpRequest;
+import org.zaproxy.addon.llm.communication.HttpRequestList;
+import org.zaproxy.addon.llm.ui.settings.LlmOptionsParam;
+import org.zaproxy.addon.llm.utils.HistoryPersister;
+import org.zaproxy.addon.llm.utils.Requestor;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class LlmCommunicationService {
+
+    private LlmOptionsParam llmOptionsParam;
+
+    private static final Logger LOGGER = LogManager.getLogger(LlmCommunicationService.class);
+    private static final String AI_REVIEWD_TAG_KEY = "AI-Reviewed";
+    private static final String AI_REVIEWD_TAG_VALUE = "1";
```

Comment:
A key can be added without a value. Though I haven't read ahead to see if this is used further.